tools/kani-cov/src/merge.rs

```diff
@@ -0,0 +1,150 @@
+// Copyright Kani Contributors
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+//! This module includes the implementation of the `merge` subcommand.
+
+use std::{
+    collections::BTreeMap,
+    fs::{File, OpenOptions},
+    io::{BufReader, BufWriter},
+    path::PathBuf,
+};
+
+use anyhow::Result;
+
+use crate::{
+    args::MergeArgs,
+    coverage::{CheckStatus, CombinedCoverageResults, CovResult, CoverageCheck, CoverageResults},
+};
+
+/// Executes the `merge` subcommand.
+///
+/// First, it loads the raw coverage results from "kaniraw" files. Then, it
+/// combines those results by coverage region thereby producing aggregated
+/// coverage information. Finally, it saves that information into another file
+/// (the coverage profile also known as the "kanicov" file).
+pub fn merge_main(args: &MergeArgs) -> Result<()> {
+    let raw_results = parse_raw_results(&args.files)?;
+    let combined_results = combine_raw_results(&raw_results);
+    save_combined_results(&combined_results, &args.output)?;
+    Ok(())
+}
+
+/// Validate arguments to the `merge` subcommand in addition to clap's
+/// validation.
+pub fn validate_merge_args(_args: &MergeArgs) -> Result<()> {
+    // No validation is done at the moment
+    Ok(())
+}
+
+/// Parse raw coverage results from a set of files (AKA "kaniraw" files)
+fn parse_raw_results(paths: &Vec<PathBuf>) -> Result<Vec<CoverageResults>> {
+    let mut raw_results = Vec::with_capacity(paths.len());
+    for path in paths {
+        let filename = path.to_string_lossy();
+        let file = File::open(path).expect(&format!("could not open file {filename}"));
+        let reader = BufReader::new(file);
+
+        let result = serde_json::from_reader(reader)
+            .expect(&format!("could not deserialize file {filename}"));
+        raw_results.push(result);
+    }
+    Ok(raw_results)
+}
+
+/// Combine raw coverage results into an aggregated form
+fn combine_raw_results(results: &Vec<CoverageResults>) -> CombinedCoverageResults {
+    let all_file_function_names = function_names_from_results(results);
+
+    let mut new_data: BTreeMap<String, Vec<(String, Vec<CovResult>)>> = BTreeMap::new();
+
+    for (file_name, fun_name) in all_file_function_names {
```

Comment:
Wouldn't references be better here?
```suggestion
    for (file_name, fun_name) in &all_file_function_names {
```

Reply:
We can use references but that forces us to use a bunch of `*` and `clone()` in the code below. I didn't think it looked better than the current solution, but happy to commit it if you think otherwise.